### PR TITLE
Add mount parallelstore service to mount parallelstore for every reboot

### PR DESCRIPTION
**Pull request by harshthakkar01**
### Submission Checklist

NOTE: Community submissions can take up to 2 weeks to be reviewed.

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cluster Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: modules/file-system/parallelstore/scripts/mount-daos.sh [hunk 1]
```diff
@@ -73,12 +74,34 @@ for i in {1..10}; do
...
 
 if ! mountpoint -q "$local_mount"; then
 	exit 1
 fi
 
+# Store the mounting logic in a variable
+mount_command='for i in {1..10}; do /bin/dfuse -m '$local_mount' --pool default-pool --container default-container --multi-user '$mount_options' --foreground && break; echo \"dfuse, failed, retrying in 1 second (attempt '$i'/10)\"; sleep 1; done'
+
+# --- Begin: Add systemd service creation ---
+cat >/usr/lib/systemd/system/mount_parallelstore.service <<EOF
```
Comment (mr0re1) [inline modules/file-system/parallelstore/scripts/mount-daos.sh:89]: Does this name should contain `$local_mount` to enable mounting multiple parallel stores?

File: modules/file-system/parallelstore/scripts/mount-daos.sh [hunk 2]
```diff
@@ -73,12 +74,34 @@ for i in {1..10}; do
...
+	echo "dfuse failed, retrying in 1 seconds (attempt $i/10)..."
 	sleep 1
 done
 
 if ! mountpoint -q "$local_mount"; then
 	exit 1
 fi
 
+# Store the mounting logic in a variable
+mount_command='for i in {1..10}; do /bin/dfuse -m '$local_mount' --pool default-pool --container default-container --multi-user '$mount_options' --foreground && break; echo \"dfuse, failed, retrying in 1 second (attempt '$i'/10)\"; sleep 1; done'
```
Comment (mr0re1) [inline modules/file-system/parallelstore/scripts/mount-daos.sh:86]: Shall we remove line 72-79 and just invoke this service once at the end?

**Review by mr0re1 — APPROVED** [review]

File: modules/file-system/parallelstore/scripts/mount-daos.sh (hunk 1 above)
Comment (harshthakkar01) [reply modules/file-system/parallelstore/scripts/mount-daos.sh:89]: ATM, mounting multiple parallelstore instance to the same cluster is not supported.

**Comment by harshthakkar01** [comment]
I do feel, this fix should be improved.

**Pull request merged**